catalog/resource_sql_table.go

```diff
@@ -294,8 +308,24 @@ func (ti *SqlTableInfo) deleteTable() error {
 
 func (ti *SqlTableInfo) applySql(sqlQuery string) error {
 	log.Printf("[INFO] Executing Sql: %s", sqlQuery)
-	r := ti.exec.Execute(ti.ClusterID, "sql", sqlQuery)
+	if ti.WarehouseID != "" {
+		execCtx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		defer cancel()
+		sqlRes, err := ti.sqlExec.ExecuteStatement(execCtx, sql.ExecuteStatementRequest{
+			Statement:   sqlQuery,
+			WaitTimeout: "10s",
```

Comment:
How is this timeout determined? What if the user's query needs more time to run?